core/src/main/java/org/apache/accumulo/core/util/ratelimit/SharedRateLimiterFactory.java

```diff
@@ -87,12 +90,12 @@ public static synchronized SharedRateLimiterFactory getInstance(AccumuloConfigur
   public RateLimiter create(String name, RateProvider rateProvider) {
     synchronized (activeLimiters) {
       if (activeLimiters.containsKey(name)) {
-        return activeLimiters.get(name);
+        return activeLimiters.get(name).get();
       } else {
         long initialRate;
         initialRate = rateProvider.getDesiredRate();
         SharedRateLimiter limiter = new SharedRateLimiter(name, rateProvider, initialRate);
-        activeLimiters.put(name, limiter);
+        activeLimiters.put(name, new WeakReference<>(limiter));
         return limiter;
```

Comment:
The typical pattern with WeakReferences in a WeakHashMap should be to check if the `weakReference.get()` is null. If it's null, it shouldn't be returned, but instead recomputed, as in the `else` clause. That is because map entries in the WeakHashMap may not be cleaned up yet, even if the WeakReference is already null.

Also, this whole if/else block may be more easily implemented inside a lambda passed to `activeLimiters.compute()`.

Reply:
5c274bf should address your suggestions @ctubbsii. I'm not sure if I understand your suggestion on using a lambda: 
 
> Also, this whole if/else block may be more easily implemented inside a lambda passed to `activeLimiters.compute()`.

Reply:
What I was referring to was the ability to use something like:

```java
  var storedValue = map.compute(key, (key, oldValue) -> {
    if (oldValue != null) {
      return oldValue; // no change
    }
    return generateNewValueForKey(key);
  });
```

It won't work in this case, because we'd be storing a WeakReference in the value, and we'd need to keep a strong reference around before returning from the lambda inside the compute method. We could do that by setting a final AtomicReference from within the compute method... but it's more complicated than just using get and put instead of compute, so it's not worth it. See my PR against your PR for a better suggestion.